src/commands/rules/addCustomFunction.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { IActionContext } from '@microsoft/vscode-azext-utils';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { ext } from '../../extensionVariables';
+import { FunctionsTreeItem } from '../../tree/rules/FunctionsTreeItem';
+import { UiStrings } from '../../uiStrings';
+
+export async function addCustomFunction(context: IActionContext, node: FunctionsTreeItem) {
+    const customFunctionName = await vscode.window.showInputBox({ title: UiStrings.InputCustomFunctionName, ignoreFocusOut: true });
+    if (!customFunctionName) {
+        return;
+    }
+
+    const sourcePathUri = vscode.Uri.file(path.join(ext.context.extensionPath, 'templates', 'rules', 'functions', 'function.js'));
+    const targetPathUri = vscode.Uri.file(path.join(node.rulesFolderPath, node.functionsDir, `${customFunctionName}.js`));
```

Comment:
what if user input like 'test.js', so this file will be like 'test.js.ts'? is it works?